File: Core kernel Yaiwes/Backend watchdog workflow adaptativo/sandbox-runtime/iii-sandbox/packages/sdk-python/src/iii_sandbox/volume.py
from __future__ import annotations

from .client import HttpClient
from .types import VolumeInfo


class VolumeManager:
    def __init__(self, client: HttpClient):
        self._client = client

    async def create(
        self, name: str, driver: str | None = None
    ) -> VolumeInfo:
        body: dict = {"name": name}
        if driver is not None:
            body["driver"] = driver
        data = await self._client.post("/sandbox/volumes", body)
        return VolumeInfo(**data)

    async def list(self) -> dict:
        return await self._client.get("/sandbox/volumes")

    async def delete(self, volume_id: str) -> dict:
        return await self._client.delete(
            f"/sandbox/volumes/{volume_id}"
        )

    async def attach(
        self,
        volume_id: str,
        sandbox_id: str,
        mount_path: str,
    ) -> dict:
        return await self._client.post(
            f"/sandbox/volumes/{volume_id}/attach",
            {"sandboxId": sandbox_id, "mountPath": mount_path},
        )

    async def detach(self, volume_id: str) -> dict:
        return await self._client.post(
            f"/sandbox/volumes/{volume_id}/detach"
        )

File: _work/backend-watchdog-full-repair-02/src/iii-sandbox/packages/sdk-rust/src/volume.rs
use std::sync::Arc;

use crate::client::HttpClient;
use crate::error::Result;
use crate::types::{
    VolumeAttachResponse, VolumeDeleteResponse, VolumeDetachResponse, VolumeInfo,
    VolumeListResponse,
};

pub struct VolumeManager {
    client: Arc<HttpClient>,
}

impl VolumeManager {
    pub fn new(client: Arc<HttpClient>) -> Self {
        Self { client }
    }

    pub async fn create(&self, name: &str, driver: Option<&str>) -> Result<VolumeInfo> {
        self.client
            .post(
                "/sandbox/volumes",
                Some(&serde_json::json!({
                    "name": name,
                    "driver": driver,
                })),
            )
            .await
    }

    pub async fn list(&self) -> Result<VolumeListResponse> {
        self.client.get("/sandbox/volumes").await
    }

    pub async fn delete(&self, volume_id: &str) -> Result<VolumeDeleteResponse> {
        self.client
            .delete(&format!("/sandbox/volumes/{volume_id}"))
            .await
    }

    pub async fn attach(
        &self,
        volume_id: &str,
        sandbox_id: &str,
        mount_path: &str,
    ) -> Result<VolumeAttachResponse> {
        self.client
            .post(
                &format!("/sandbox/volumes/{volume_id}/attach"),
                Some(&serde_json::json!({
                    "sandboxId": sandbox_id,
                    "mountPath": mount_path,
                })),
            )
            .await
    }

    pub async fn detach(&self, volume_id: &str) -> Result<VolumeDetachResponse> {
        self.client
            .post_no_body(&format!("/sandbox/volumes/{volume_id}/detach"))
            .await
    }
}
